AsyncDisplayKit/Details/_ASDisplayViewAccessiblity.mm

```diff
@@ -0,0 +1,122 @@
+/* Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "_ASDisplayViewAccessiblity.h"
+#import "_ASDisplayView.h"
+#import "ASDisplayNodeExtras.h"
+#import "ASDisplayNode+FrameworkPrivate.h"
+
+#import <objc/runtime.h>
+
+
+#pragma mark - UIAccessibilityElement
+
+@implementation UIAccessibilityElement (_ASDisplayView)
+
++ (UIAccessibilityElement *)accessibilityElementWithContainer:(id)container node:(ASDisplayNode *)node
+{
+  UIAccessibilityElement *accessibilityElement = [[UIAccessibilityElement alloc] initWithAccessibilityContainer:container];
+  accessibilityElement.accessibilityIdentifier = node.accessibilityIdentifier;
+  accessibilityElement.accessibilityLabel = node.accessibilityLabel;
+  accessibilityElement.accessibilityHint = node.accessibilityHint;
+  accessibilityElement.accessibilityValue = node.accessibilityValue;
+  accessibilityElement.accessibilityTraits = node.accessibilityTraits;
+  return accessibilityElement;
+}
+
+@end
+
+
+#pragma mark - _ASDisplayView / UIAccessibilityContainer
+
+@interface _ASDisplayView () {
+  NSMutableArray *_accessibleElements;
+}
+
+@end
+
+@implementation _ASDisplayView (UIAccessibilityContainer)
+
+#pragma mark - UIAccessibility
+
+- (NSArray *)accessibleElements
+{
+  ASDisplayNode *selfNode = self.asyncdisplaykit_node;
+  if (selfNode == nil) {
+    return nil;
+  }
+  
+  _accessibleElements = [[NSMutableArray alloc] init];
```

Comment:
Let's make the ivar NSArray *, and create a local NSMutableArray * here.  It helps show that there are no areas that are allowed to edit teh array outside this method.